feat(AVZP-66): BING_API_KEY setting is now configurable from values.yaml

phardy-egis opened the pull request:
## Description

Currently the HELM does not allow to configure the following environment variables useful for Django Backend:
- django BING_API_KEY setting to enable tthis feature as part of MapStore
- django SECRET_KEY setting to enable secret rotation

This PR is a proposal to add these settings.

## Type of Change

New Feature

## Checklist

Please ensure that your pull request meets the following requirements:

- The pull request is limited to one type (docs, feature, bug fix, etc.)
- The pull request is as small as possible. Consider opening multiple pull requests instead of one large one.
- The feature or bug fix has been discussed and documented in an issue beforehand.

## Additional Notes

Any additional information or context regarding the pull request can be provided here.

Maybe documentation related to this feature should be improved

mwallschlaeger reviewed (approved):
yes resonable addition. will be included into the next release

mwallschlaeger reviewed (changes requested):
linting requires, that `.Values.geonode.secret.django.secretKey` and `.Values.geonode.secret.bing.apiKey` must have default values in values.yaml. Could you add them?

phardy-egis commented:
@mwallschlaeger 

I added default values according to the default ones provided in the geonode repo:
- SECRET_KEY: https://github.com/GeoNode/geonode/blob/master/.env_local#L167C13-L167C63
- BING_API_KEY: https://github.com/GeoNode/geonode/blob/master/.env_local#L192

Also updated documentation.

mwallschlaeger reviewed (approved):
Cool thanks for the PR

phardy-egis commented:
I noticed CI job failed. Let me know if corrections should be done on my side:
https://github.com/GeoNodeUserGroup-DE/geonode-k8s/actions/runs/13432035876/job/37525847702
